geonode/base/api/views.py

```diff
@@ -1415,6 +1419,85 @@ def linked_resources(self, request, pk, *args, **kwargs):
 
         return base_linked_resources(self.get_object().get_real_instance(), request.user, request.GET)
 
+    @action(
+        detail=True,
+        methods=["post"],
+        url_path=r"assets",
+        permission_classes=[UserHasPerms(perms_dict={"default": {"POST": ["base.add_resourcebase"]}})],
+        parser_classes=[JSONParser, MultiPartParser],
+        url_name="assets",
+    )
+    def asset(self, request, pk=None, *args, **kwargs):
+        """Handles uploading a new file, creating an asset, and linking it to the resource."""
+
+        if "file" not in request.FILES:
+            return Response({"message": "File not provided."}, status=status.HTTP_400_BAD_REQUEST)
+        file = request.FILES["file"]
+        resource = get_object_or_404(ResourceBase, pk=pk)
+        user = request.user
+        title = request.data.get("title", None)
+        description = request.data.get("description", None)
+
+        forbidden_titles = {"Original", "Files"}
+        if title in forbidden_titles:
+            return Response(
+                {"message": f"'{title}' is a reserved title and cannot be used."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        if not permissions_registry.user_has_perm(user, resource, "change_resourcebase", include_virtual=True):
+            return Response(
+                {"message": "You do not have permission to add an asset to this resource."},
+                status=status.HTTP_403_FORBIDDEN,
+            )
+
+        if file and not os.path.splitext(file.name)[1].lower()[1:] in settings.ALLOWED_DOCUMENT_TYPES:
+            logger.debug("This file type is not allowed")
+            return Response({"message": "The uploaded file type is not allowed."}, status=status.HTTP_400_BAD_REQUEST)
+        try:
+            handler = asset_handler_registry.get_default_handler()
+            asset, link = create_asset_and_link(
+                resource, user, [file], handler=handler, title=title, description=description, clone_files=False
+            )
+            if asset and link:
+                return Response(
+                    {"message": "Asset created and linked successfully.", "asset_id": asset.id},
```

Comment:
i would also return the `link_id`, since it has just been created as an effect of the POST call